src/appleseed/renderer/kernel/rendering/pixelrendererbase.h

```diff
@@ -32,20 +32,26 @@
 
 // appleseed.renderer headers.
 #include "renderer/kernel/rendering/ipixelrenderer.h"
+#include "renderer/modeling/frame/frame.h"
 
 // appleseed.foundation headers.
+#include "foundation/image/tile.h"
 #include "foundation/math/vector.h"
 #include "foundation/platform/compiler.h"
 
 // Standard headers.
 #include <cstddef>
+#include <memory>
 
 // Forward declarations.
 namespace foundation    { class Tile; }
 namespace renderer      { class AOVAccumulatorContainer; }
 namespace renderer      { class Frame; }
 namespace renderer      { class TileStack; }
 
+using namespace foundation;
```

Comment:
We completely forbid `using` declarations in header files since this defeats the point of namespaces entirely. Please use fully qualified names.

Reply:
That makes sense, we should change that also on the adaptive pixel renderer